Does not build with mysql

`my_bool` defined in src/engine/server/databases/mysql.cpp conflicts with typedef from mysql. mysql is 5.7.

```
/usr/local/libexec/ccache/c++ -DCONF_INFORM_UPDATE -DCONF_MYSQL -DCONF_OPENSSL -DCONF_UPNP -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"16.2\" -DGLEW_STATIC -D_FORTIFY_SOURCE=2 -I/work/usr/ports/games/DDNet/work/.build/src -I/work/usr/ports/games/DDNet/work/DDNet-16.2/src -isystem /usr/local/include -isystem /usr/local/include/miniupnpc -isystem /usr/local/include/mysql -O2 -pipe -fstack-protector-strong -fno-strict-aliasing  -DBINARY_DIR=\"/usr/local/bin\" -O2 -pipe -fstack-protector-strong -fno-strict-aliasing  -DBINARY_DIR=\"/usr/local/bin\" -fdiagnostics-color=always -fcolor-diagnostics -fstack-protector-all -fno-exceptions -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wshadow-all -Wthread-safety -Wthread-safety-negative -Wsuggest-override -std=c++17 -MD -MT CMakeFiles/DDNet-Server.dir/src/engine/server/databases/mysql.cpp.o -MF CMakeFiles/DDNet-Server.dir/src/engine/server/databases/mysql.cpp.o.d -o CMakeFiles/DDNet-Server.dir/src/engine/server/databases/mysql.cpp.o -c /work/usr/ports/games/DDNet/work/DDNet-16.2/src/engine/server/databases/mysql.cpp
/work/usr/ports/games/DDNet/work/DDNet-16.2/src/engine/server/databases/mysql.cpp:14:14: error: typedef redefinition with different types ('bool' vs 'char')
typedef bool my_bool;
             ^
/usr/local/include/mysql/mysql.h:49:14: note: previous definition is here
typedef char my_bool;
             ^
1 error generated.
```

--- a/src/engine/server/databases/mysql.cpp
+++ b/src/engine/server/databases/mysql.cpp
@@ -10,7 +10,8 @@
 #include <memory>
 #include <vector>
 
-#ifndef LIBMARIADB
+// MySQL >= 8.0.1 removed my_bool, 8.0.2 accidentally reintroduced it: https://bugs.mysql.com/bug.php?id=87337
+#if !defined(LIBMARIADB) && MYSQL_VERSION_ID >= 80001 && MYSQL_VERSION_ID != 80002
 typedef bool my_bool;
 #endif
 

